stratosphere/ams_mitm/source/utils.cpp

```diff
@@ -665,3 +667,88 @@ Result Utils::GetSettingsItemBooleanValue(const char *name, const char *key, boo
 void Utils::RebootToFatalError(AtmosphereFatalErrorContext *ctx) {
     BpcRebootManager::RebootForFatalError(ctx);
 }
+
+void Utils::CreateBlankProdinfoIfNeeded() {
+    /* if (!Utils::HasGlobalFlag("blank_prodinfo")) {
```

Comment:
This will never work because the SD card is not initialized at the time you're doing this. But you want to do it unconditionally anyway, so just remove it.

Reply:
Why do the fsFsCreateFile calls work then?

Reply:
```
bool Utils::HasGlobalFlag(const char *flag) {
    if (IsSdInitialized()) {
```

```
bool Utils::IsSdInitialized() {
    return g_has_initialized;
}```

`g_has_initialized` is not set until the end of the function that calls the code you're writing.